database/main.go

```
@@ -108,11 +120,38 @@ func saveWorkspaceDB(
 	}
 
 	if err := g.WriteFileInWorkspace(ctx, dbWorkspacePath, updatedDBData, gptscript.WriteFileInWorkspaceOptions{
-		WorkspaceID: workspaceID,
+		WorkspaceID:      workspaceID,
+		CreateRevision:   &([]bool{true}[0]),
+		LatestRevisionID: revisionID,
 	}); err != nil {
 		return fmt.Errorf("Error writing updated DB file to workspace: %v", err)
 	}
 
+	// Delete old revisions after successfully writing the new revision
+	revisions, err := g.ListRevisionsForFileInWorkspace(ctx, dbWorkspacePath, gptscript.ListRevisionsForFileInWorkspaceOptions{
+		WorkspaceID: workspaceID,
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error listing revisions: %v\n", err)
```

Do we want to intentionally ignore error here?

---

Yes, I don't think deleting old revisions should be a critical error. IIUC, this will return the error to Obot, but not the LLM. Correct me if I'm wrong there though.